libs/server-sdk/src/data_systems/lazy_load/lazy_load_system.cpp

```diff
@@ -20,15 +28,17 @@ void LazyLoad::Shutdown() {}
 
 std::shared_ptr<data_model::FlagDescriptor> LazyLoad::GetFlag(
     std::string const& key) const {
-    auto const state = tracker_.State(Keys::kAllSegments, time_());
+    auto const state =
+        tracker_.State(data_components::DataKind::kFlag, key, time_());
     return Get<std::shared_ptr<data_model::FlagDescriptor>>(
         state, [this, &key]() { RefreshFlag(key); },
         [this, &key]() { return cache_.GetFlag(key); });
 }
 
 std::shared_ptr<data_model::SegmentDescriptor> LazyLoad::GetSegment(
     std::string const& key) const {
-    auto const state = tracker_.State(Keys::kAllSegments, time_());
+    auto const state =
+        tracker_.State(data_components::DataKind::kSegment, key, time_());
     return Get<std::shared_ptr<data_model::SegmentDescriptor>>(
```

Comment:
todo: Here I thought another typo - should be checking the state of that particular segment (?)